src/kernels/jupyter/launcher/jupyterServerConnector.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { inject, injectable, optional } from 'inversify';
+import { IPythonExtensionChecker } from '../../../platform/api/types';
+import { ConnectNotebookProviderOptions, IJupyterConnection, IJupyterServerConnector } from '../../types';
+import { DisplayOptions } from '../../displayOptions';
+import { IRawKernelConnectionSessionCreator } from '../../raw/types';
+import { IJupyterNotebookProvider } from '../types';
+import { PythonExtensionNotInstalledError } from '../../../platform/errors/pythonExtNotInstalledError';
+
+@injectable()
+export class JupyterServerConnector implements IJupyterServerConnector {
```

Comment:
These are not the final names and final classes, the primary goal is to break down larger components into smaller ones , this way re-factoring this would be simpler and its easier
Right now there's a lot of inheritance and coupling